File: android/app/src/main/java/com/liwidale/liauth/ui/Components.kt
package com.liwidale.liauth.ui

import androidx.compose.foundation.Canvas
import androidx.compose.foundation.background
import androidx.compose.foundation.border
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.PaddingValues
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.height
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.size
import androidx.compose.material3.Button
import androidx.compose.material3.ButtonDefaults
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.OutlinedTextField
import androidx.compose.material3.OutlinedTextFieldDefaults
import androidx.compose.material3.Switch
import androidx.compose.material3.SwitchDefaults
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.runtime.getValue
import androidx.compose.runtime.mutableStateOf
import androidx.compose.runtime.remember
import androidx.compose.runtime.setValue
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.geometry.Size
import androidx.compose.ui.graphics.asImageBitmap
import androidx.compose.ui.graphics.luminance
import androidx.compose.ui.text.withStyle
import androidx.compose.ui.text.input.PasswordVisualTransformation
import androidx.compose.ui.text.input.VisualTransformation
import androidx.compose.ui.text.style.TextAlign
import androidx.compose.ui.unit.Dp
import androidx.compose.ui.unit.dp
import com.liwidale.liauth.ui.theme.ControlShape
import com.liwidale.liauth.ui.theme.LocalPalette

@Composable
fun PrimaryButton(
    text: String,
    onClick: () -> Unit,
    modifier: Modifier = Modifier,
    enabled: Boolean = true,
) {
    val palette = LocalPalette.current
    Button(
        onClick = onClick,
        enabled = enabled,
        shape = ControlShape,
        colors = ButtonDefaults.buttonColors(
            containerColor = palette.accent,
            contentColor = palette.accentText,
            disabledContainerColor = palette.surfaceRaised,
            disabledContentColor = palette.textTertiary,
        ),
        contentPadding = PaddingValues(horizontal = 16.dp, vertical = 0.dp),
        modifier = modifier
            .fillMaxWidth()
            .height(40.dp),
    ) {
        Text(text, style = MaterialTheme.typography.labelLarge)
    }
}

@Composable
fun SecondaryButton(
    text: String,
    onClick: () -> Unit,
    modifier: Modifier = Modifier,
) {
    val palette = LocalPalette.current
    Button(
        onClick = onClick,
        shape = ControlShape,
        colors = ButtonDefaults.buttonColors(
            containerColor = palette.surfaceRaised,
            contentColor = palette.textPrimary,
        ),
        contentPadding = PaddingValues(horizontal = 16.dp, vertical = 0.dp),
        modifier = modifier
            .fillMaxWidth()
            .height(40.dp)
            .border(1.dp, palette.border, ControlShape),
    ) {
        Text(text, style = MaterialTheme.typography.labelLarge)
    }
}

@Composable
fun LiAuthTextField(
    value: String,
    onValueChange: (String) -> Unit,
    hint: String,
    modifier: Modifier = Modifier,
    password: Boolean = false,
) {
    val palette = LocalPalette.current
    var revealed by remember { mutableStateOf(false) }
    OutlinedTextField(
        value = value,
        onValueChange = onValueChange,
        placeholder = { Text(hint, color = palette.textTertiary) },
        singleLine = true,
        visualTransformation = if (password && !revealed) {
            PasswordVisualTransformation()
        } else {
            VisualTransformation.None
        },
        trailingIcon = if (password) {
            {
                androidx.compose.material3.IconButton(onClick = { revealed = !revealed }) {
                    androidx.compose.material3.Icon(
                        if (revealed) LiAuthIcons.EyeOff else LiAuthIcons.Eye,
                        contentDescription = null,
                        tint = palette.textSecondary,
                    )
                }
            }
        } else {
            null
        },
        shape = ControlShape,
        colors = OutlinedTextFieldDefaults.colors(
            focusedContainerColor = palette.surfaceRaised,
            unfocusedContainerColor = palette.surfaceRaised,
            focusedBorderColor = palette.borderStrong,
            unfocusedBorderColor = palette.border,
            focusedTextColor = palette.textPrimary,
            unfocusedTextColor = palette.textPrimary,
            cursorColor = palette.textPrimary,
        ),
        modifier = modifier.fillMaxWidth(),
    )
}

@Composable
fun LiAuthSwitch(checked: Boolean, onCheckedChange: (Boolean) -> Unit) {
    val palette = LocalPalette.current
    Switch(
        checked = checked,
        onCheckedChange = onCheckedChange,
        colors = SwitchDefaults.colors(
            checkedThumbColor = palette.accentText,
            checkedTrackColor = palette.accent,
            uncheckedThumbColor = palette.textSecondary,
            uncheckedTrackColor = palette.surfaceRaised,
            uncheckedBorderColor = palette.borderStrong,
        ),
    )
}

@Composable
fun CountdownBar(fraction: Float, modifier: Modifier = Modifier) {
    val palette = LocalPalette.current
    Canvas(
        modifier = modifier
            .fillMaxWidth()
            .height(2.dp),
    ) {
        drawRect(color = palette.border)
        drawRect(
            color = palette.textPrimary,
            size = Size(size.width * fraction.coerceIn(0f, 1f), size.height),
        )
    }
}

/** Primary brand colors for well-known services, keyed by normalized issuer. */
object BrandColors {
    private val colors: Map<String, androidx.compose.ui.graphics.Color> = buildMap {
        fun put(name: String, rgb: Long) = put(name, androidx.compose.ui.graphics.Color(0xFF000000 or rgb))
        put("github", 0x181717); put("gitlab", 0xFC6D26); put("google", 0x4285F4)
        put("gmail", 0x4285F4); put("youtube", 0xFF0000); put("microsoft", 0x0078D4)
        put("outlook", 0x0078D4); put("azure", 0x0078D4); put("apple", 0x0A84FF)
        put("icloud", 0x0A84FF); put("amazon", 0xFF9900); put("aws", 0xFF9900)
        put("facebook", 0x1877F2); put("meta", 0x1877F2); put("instagram", 0xE1306C)
        put("x", 0x1D9BF0); put("twitter", 0x1D9BF0); put("discord", 0x5865F2)
        put("slack", 0x4A154B); put("dropbox", 0x0061FF); put("steam", 0x1B2838)
        put("epicgames", 0x2C2C2C); put("riotgames", 0xEB0014); put("twitch", 0x9146FF)
        put("reddit", 0xFF4500); put("linkedin", 0x0A66C2); put("paypal", 0x003087)
        put("stripe", 0x635BFF); put("coinbase", 0x0052FF); put("binance", 0xF0B90B)
        put("kraken", 0x5729CE); put("protonmail", 0x6D4AFF); put("proton", 0x6D4AFF)
        put("bitwarden", 0x175DDC); put("nextcloud", 0x0082C9); put("cloudflare", 0xF38020)
        put("digitalocean", 0x0069FF); put("npm", 0xCB3837); put("docker", 0x2496ED)
        put("atlassian", 0x0052CC); put("jira", 0x0052CC); put("bitbucket", 0x0052CC)
        put("notion", 0x000000); put("figma", 0xA259FF); put("telegram", 0x24A1DE)
        put("whatsapp", 0x25D366); put("signal", 0x3A76F0); put("wordpress", 0x21759B)
        put("shopify", 0x5FBE42); put("ebay", 0xE63238); put("netflix", 0xE50914)
        put("spotify", 0x1ED760); put("nintendo", 0xE60012); put("playstation", 0x00439C)
        put("xbox", 0x107C10); put("yandex", 0xFFCC00); put("vk", 0x0077FF)
        put("mailru", 0x005FF9); put("heroku", 0x430098); put("netlify", 0x00ADB5)
        put("vercel", 0x000000); put("lastpass", 0xD52B1E); put("1password", 0x0A84FF)
    }

    fun forIssuer(issuer: String): androidx.compose.ui.graphics.Color? {
        val normalized = issuer.lowercase().filter { it.isLetterOrDigit() }
        return colors[normalized]
    }

    /// Simple Icons slugs bundled under assets/icons (white 96x96 PNGs, CC0).
    private val iconSlugs = setOf(
        "github", "gitlab", "google", "gmail", "youtube", "apple", "icloud",
        "facebook", "meta", "instagram", "x", "discord", "dropbox", "steam",
        "epicgames", "riotgames", "twitch", "reddit", "paypal", "stripe",
        "coinbase", "binance", "protonmail", "proton", "bitwarden", "nextcloud",
        "cloudflare", "digitalocean", "npm", "docker", "atlassian", "jira",
        "bitbucket", "notion", "figma", "telegram", "whatsapp", "signal",
        "wordpress", "shopify", "ebay", "netflix", "spotify", "playstation",
        "vk", "maildotru", "netlify", "vercel", "lastpass", "1password",
    )

    fun iconSlug(issuer: String): String? {
        val normalized = issuer.lowercase().filter { it.isLetterOrDigit() }
        val slug = when (normalized) {
            "twitter" -> "x"
            "mailru", "mail" -> "maildotru"
            else -> normalized
        }
        return slug.takeIf { it in iconSlugs }
    }
}

@Composable
fun Avatar(title: String, size: Dp, modifier: Modifier = Modifier, branded: Boolean = false) {
    val palette = LocalPalette.current
    val brand = if (branded) BrandColors.forIssuer(title) else null
    val slug = if (branded) BrandColors.iconSlug(title) else null

    if (slug != null) {
        // Real Simple Icons logo (white glyph) on the brand background.
        val context = androidx.compose.ui.platform.LocalContext.current
        val bitmap = remember(slug) {
            runCatching {
                context.assets.open("icons/$slug.png").use {
                    android.graphics.BitmapFactory.decodeStream(it)
                }
            }.getOrNull()
        }
        if (bitmap != null) {
            val background = brand ?: androidx.compose.ui.graphics.Color(0xFF1C1C1C)
            Box(
                contentAlignment = Alignment.Center,
                modifier = modifier
                    .size(size)
                    .background(background, ControlShape),
            ) {
                androidx.compose.foundation.Image(
                    bitmap = bitmap.asImageBitmap(),
                    contentDescription = title,
                    modifier = Modifier.size(size * 0.6f),
                )
            }
            return
        }
    }

    val background = brand ?: palette.surfaceRaised
    val borderColor = brand ?: palette.borderStrong
    val textColor = when {
        brand == null -> palette.textPrimary
        brand.luminance() > 0.6f -> androidx.compose.ui.graphics.Color.Black
        else -> androidx.compose.ui.graphics.Color.White
    }
    Box(
        contentAlignment = Alignment.Center,
        modifier = modifier
            .size(size)
            .background(background, ControlShape)
            .border(1.dp, borderColor, ControlShape),
    ) {
        Text(
            text = title.firstOrNull()?.uppercase() ?: "",
            style = MaterialTheme.typography.titleLarge,
            color = textColor,
            textAlign = TextAlign.Center,
        )
    }
}

/**
 * Text with the characters at [indices] (char positions from the fuzzy
 * matcher) rendered inverted, used to highlight search hits.
 */
@Composable
fun HighlightedText(
    text: String,
    indices: List<UInt>,
    style: androidx.compose.ui.text.TextStyle,
    color: androidx.compose.ui.graphics.Color,
) {
    val palette = LocalPalette.current
    if (indices.isEmpty()) {
        Text(text, style = style, color = color)
        return
    }
    val matched = indices.map { it.toInt() }.toSet()
    val annotated = androidx.compose.ui.text.buildAnnotatedString {
        text.forEachIndexed { i, ch ->
            if (i in matched) {
                withStyle(
                    androidx.compose.ui.text.SpanStyle(
                        background = palette.accent,
                        color = palette.accentText,
                    ),
                ) { append(ch) }
            } else {
                append(ch)
            }
        }
    }
    Text(annotated, style = style, color = color)
}

@Composable
fun PinMarker(modifier: Modifier = Modifier) {
    val palette = LocalPalette.current
    Box(
        modifier = modifier
            .size(6.dp)
            .background(palette.textSecondary, androidx.compose.foundation.shape.CircleShape),
    )
}

@Composable
fun SectionLabel(text: String, modifier: Modifier = Modifier) {
    val palette = LocalPalette.current
    Text(
        text = text.uppercase(),
        style = MaterialTheme.typography.labelSmall,
        color = palette.textTertiary,
        modifier = modifier.padding(bottom = 8.dp),
    )
}

fun formatCode(code: String): String = when (code.length) {
    6 -> "${code.substring(0, 3)} ${code.substring(3)}"
    8 -> "${code.substring(0, 4)} ${code.substring(4)}"
    else -> code
}

File: apple/Sources/UI/Components.swift
import SwiftUI

#if os(macOS)
import AppKit
#else
import UIKit
#endif

struct PrimaryButton: View {
    let title: String
    var enabled: Bool = true
    let action: () -> Void

    var body: some View {
        Button(action: action) {
            Text(title)
                .font(.inter(14, weight: .semibold))
                .foregroundColor(enabled ? Palette.accentText : Palette.textTertiary)
                .frame(maxWidth: .infinity)
                .frame(height: 40)
                .background(
                    RoundedRectangle(cornerRadius: Radius.control)
                        .fill(enabled ? Palette.accent : Palette.surfaceRaised)
                )
        }
        .buttonStyle(.plain)
        .disabled(!enabled)
    }
}

struct SecondaryButton: View {
    let title: String
    let action: () -> Void

    var body: some View {
        Button(action: action) {
            Text(title)
                .font(.inter(14, weight: .medium))
                .foregroundColor(Palette.textPrimary)
                .frame(maxWidth: .infinity)
                .frame(height: 40)
                .background(RoundedRectangle(cornerRadius: Radius.control).fill(Palette.surfaceRaised))
                .overlay(RoundedRectangle(cornerRadius: Radius.control).stroke(Palette.border, lineWidth: 1))
        }
        .buttonStyle(.plain)
    }
}

struct LiAuthField: View {
    let hint: String
    @Binding var text: String
    var secure: Bool = false

    @State private var revealed = false
    @FocusState private var focused: Bool

    var body: some View {
        HStack(spacing: 8) {
            Group {
                if secure && !revealed {
                    SecureField(hint, text: $text)
                } else {
                    TextField(hint, text: $text)
                }
            }
            .font(.inter(15))
            .textFieldStyle(.plain)
            .foregroundColor(Palette.textPrimary)
            #if os(iOS)
            .autocorrectionDisabled()
            .textInputAutocapitalization(.never)
            #endif

            if secure {
                Button {
                    revealed.toggle()
                } label: {
                    Image(systemName: revealed ? "eye.slash" : "eye")
                        .font(.system(size: 15))
                        .foregroundColor(Palette.textSecondary)
                }
                .buttonStyle(.plain)
                .accessibilityLabel(revealed ? "Hide" : "Show")
            }
        }
        .padding(.horizontal, 12)
        .frame(height: 40)
        .background(RoundedRectangle(cornerRadius: Radius.control).fill(Palette.surface))
        .overlay(
            RoundedRectangle(cornerRadius: Radius.control)
                .stroke(focused ? Palette.textPrimary : Palette.border, lineWidth: 1)
        )
        .focused($focused)
    }
}

struct SectionLabel: View {
    let text: String

    var body: some View {
        Text(text.uppercased())
            .font(.inter(12, weight: .medium))
            .kerning(0.8)
            .foregroundColor(Palette.textTertiary)
    }
}

struct CountdownBar: View {
    let fraction: Double

    var body: some View {
        GeometryReader { geometry in
            ZStack(alignment: .leading) {
                Rectangle().fill(Palette.border)
                Rectangle()
                    .fill(fraction < 0.2 ? Palette.warning : Palette.textPrimary)
                    .frame(width: geometry.size.width * max(0, min(1, fraction)))
            }
        }
        .frame(height: 2)
    }
}

/// Primary brand colors for well-known services, keyed by normalized issuer.
enum BrandColors {
    private static let colors: [String: (Double, Double, Double)] = [
        "github": (24, 23, 23), "gitlab": (252, 109, 38), "google": (66, 133, 244),
        "gmail": (66, 133, 244), "youtube": (255, 0, 0), "microsoft": (0, 120, 212),
        "outlook": (0, 120, 212), "azure": (0, 120, 212), "apple": (10, 132, 255),
        "icloud": (10, 132, 255), "amazon": (255, 153, 0), "aws": (255, 153, 0),
        "facebook": (24, 119, 242), "meta": (24, 119, 242), "instagram": (225, 48, 108),
        "x": (29, 155, 240), "twitter": (29, 155, 240), "discord": (88, 101, 242),
        "slack": (74, 21, 75), "dropbox": (0, 97, 255), "steam": (27, 40, 56),
        "epicgames": (44, 44, 44), "riotgames": (235, 0, 20), "twitch": (145, 70, 255),
        "reddit": (255, 69, 0), "linkedin": (10, 102, 194), "paypal": (0, 48, 135),
        "stripe": (99, 91, 255), "coinbase": (0, 82, 255), "binance": (240, 185, 11),
        "kraken": (87, 41, 206), "protonmail": (109, 74, 255), "proton": (109, 74, 255),
        "bitwarden": (23, 93, 220), "nextcloud": (0, 130, 201), "cloudflare": (243, 128, 32),
        "digitalocean": (0, 105, 255), "npm": (203, 56, 55), "docker": (36, 150, 237),
        "atlassian": (0, 82, 204), "jira": (0, 82, 204), "bitbucket": (0, 82, 204),
        "notion": (0, 0, 0), "figma": (162, 89, 255), "telegram": (36, 161, 222),
        "whatsapp": (37, 211, 102), "signal": (58, 118, 240), "wordpress": (33, 117, 155),
        "shopify": (95, 190, 66), "ebay": (230, 50, 56), "netflix": (229, 9, 20),
        "spotify": (30, 215, 96), "nintendo": (230, 0, 18), "playstation": (0, 67, 156),
        "xbox": (16, 124, 16), "yandex": (255, 204, 0), "vk": (0, 119, 255),
        "mailru": (0, 95, 249), "heroku": (67, 0, 152), "netlify": (0, 173, 181),
        "vercel": (0, 0, 0), "lastpass": (213, 43, 30), "1password": (10, 132, 255),
    ]

    static func forIssuer(_ issuer: String) -> Color? {
        let normalized = issuer.lowercased().filter { $0.isLetter || $0.isNumber }
        guard let rgb = colors[String(normalized)] else { return nil }
        return Color(red: rgb.0 / 255, green: rgb.1 / 255, blue: rgb.2 / 255)
    }

    static func isLight(_ issuer: String) -> Bool {
        let normalized = issuer.lowercased().filter { $0.isLetter || $0.isNumber }
        guard let rgb = colors[String(normalized)] else { return false }
        return 0.299 * rgb.0 + 0.587 * rgb.1 + 0.114 * rgb.2 > 150
    }

    /// Simple Icons slugs bundled under icons/ (white 96x96 PNGs, CC0).
    private static let iconSlugs: Set<String> = [
        "github", "gitlab", "google", "gmail", "youtube", "apple", "icloud",
        "facebook", "meta", "instagram", "x", "discord", "dropbox", "steam",
        "epicgames", "riotgames", "twitch", "reddit", "paypal", "stripe",
        "coinbase", "binance", "protonmail", "proton", "bitwarden", "nextcloud",
        "cloudflare", "digitalocean", "npm", "docker", "atlassian", "jira",
        "bitbucket", "notion", "figma", "telegram", "whatsapp", "signal",
        "wordpress", "shopify", "ebay", "netflix", "spotify", "playstation",
        "vk", "maildotru", "netlify", "vercel", "lastpass", "1password",
    ]

    static func iconSlug(_ issuer: String) -> String? {
        let normalized = String(issuer.lowercased().filter { $0.isLetter || $0.isNumber })
        let slug: String
        switch normalized {
        case "twitter": slug = "x"
        case "mailru", "mail": slug = "maildotru"
        default: slug = normalized
        }
        return iconSlugs.contains(slug) ? slug : nil
    }

    static func icon(_ issuer: String) -> Image? {
        guard let slug = iconSlug(issuer),
              let url = Bundle.main.url(forResource: slug, withExtension: "png", subdirectory: "icons")
        else {
            return nil
        }
        #if os(macOS)
        guard let image = NSImage(contentsOf: url) else { return nil }
        return Image(nsImage: image)
        #else
        guard let data = try? Data(contentsOf: url), let image = UIImage(data: data) else { return nil }
        return Image(uiImage: image)
        #endif
    }
}

struct AvatarView: View {
    let title: String
    var size: CGFloat = 42
    var branded: Bool = false

    var body: some View {
        let brand = branded ? BrandColors.forIssuer(title) : nil
        let logo = branded ? BrandColors.icon(title) : nil
        ZStack {
            if let logo {
                // Real Simple Icons logo (white glyph) on the brand background.
                RoundedRectangle(cornerRadius: Radius.control)
                    .fill(brand ?? Color(red: 0.11, green: 0.11, blue: 0.11))
                logo
                    .resizable()
                    .aspectRatio(contentMode: .fit)
                    .frame(width: size * 0.6, height: size * 0.6)
            } else {
                RoundedRectangle(cornerRadius: Radius.control)
                    .fill(brand ?? Palette.surfaceRaised)
                    .overlay(
                        RoundedRectangle(cornerRadius: Radius.control)
                            .stroke(brand ?? Palette.border, lineWidth: 1)
                    )
                Text(title.prefix(1).uppercased())
                    .font(.inter(size * 0.4, weight: .semibold))
                    .foregroundColor(
                        brand == nil
                            ? Palette.textPrimary
                            : (BrandColors.isLight(title) ? .black : .white)
                    )
            }
        }
        .frame(width: size, height: size)
    }
}

/// Text with the characters at `indices` (char positions from the fuzzy
/// matcher) rendered inverted, used to highlight search hits.
struct HighlightedText: View {
    let text: String
    let indices: [UInt32]
    let font: Font
    let color: Color

    var body: some View {
        if indices.isEmpty {
            Text(text).font(font).foregroundColor(color)
        } else {
            Text(attributed).font(font)
        }
    }

    private var attributed: AttributedString {
        var result = AttributedString()
        let matched = Set(indices.map { Int($0) })
        for (i, character) in text.enumerated() {
            var piece = AttributedString(String(character))
            if matched.contains(i) {
                piece.backgroundColor = Palette.accent
                piece.foregroundColor = Palette.accentText
            } else {
                piece.foregroundColor = color
            }
            result += piece
        }
        return result
    }
}

struct LogoMark: View {
    var size: CGFloat

    var body: some View {
        if let image = loadBundledImage(named: "logo") {
            image
                .renderingMode(.template)
                .resizable()
                .aspectRatio(contentMode: .fit)
                .frame(width: size, height: size)
                .foregroundColor(Palette.textPrimary)
        } else {
            Rectangle()
                .stroke(Palette.borderStrong, lineWidth: 1.5)
                .frame(width: size * 0.85, height: size * 0.85)
        }
    }
}

func loadBundledImage(named name: String) -> Image? {
    guard let url = Bundle.main.url(forResource: name, withExtension: "png") else { return nil }
    #if os(macOS)
    guard let image = NSImage(contentsOf: url) else { return nil }
    return Image(nsImage: image)
    #else
    guard let data = try? Data(contentsOf: url), let image = UIImage(data: data) else { return nil }
    return Image(uiImage: image)
    #endif
}

struct Wordmark: View {
    var height: CGFloat

    var body: some View {
        if let image = Self.load() {
            image
                .renderingMode(.template)
                .resizable()
                .aspectRatio(contentMode: .fit)
                .frame(height: height)
                .foregroundColor(Palette.textPrimary)
                .accessibilityLabel("LiAuth")
        } else {
            Text("LiAuth")
                .font(.interBold(height * 0.9))
                .foregroundColor(Palette.textPrimary)
        }
    }

    private static func load() -> Image? {
        loadBundledImage(named: "text")
    }
}

struct PinMarker: View {
    var body: some View {
        Circle()
            .fill(Palette.textSecondary)
            .frame(width: 6, height: 6)
    }
}

struct Chip: View {
    let text: String
    let selected: Bool
    let action: () -> Void

    var body: some View {
        Button(action: action) {
            Text(text)
                .font(.inter(13, weight: .medium))
                .foregroundColor(selected ? Palette.accentText : Palette.textSecondary)
                .padding(.horizontal, 12)
                .padding(.vertical, 6)
                .background(
                    RoundedRectangle(cornerRadius: Radius.control)
                        .fill(selected ? Palette.accent : Palette.surfaceRaised)
                )
                .overlay(
                    RoundedRectangle(cornerRadius: Radius.control)
                        .stroke(selected ? Color.clear : Palette.border, lineWidth: 1)
                )
        }
        .buttonStyle(.plain)
    }
}
